src/librustc_target/spec/wasm32_wasi.rs

```diff
@@ -86,7 +86,10 @@ pub fn target() -> Result<Target, String> {
 
     // When generating an executable be sure to put the startup object at the
     // front so the main function is correctly hooked up.
-    options.pre_link_objects_exe_crt.push("crt1.o".to_string());
+    options.pre_link_objects = crt_objects::new(&[
+        (CrtObjectKind::StaticNoPicExe, &["crt1.o"]),
+        (CrtObjectKind::StaticPicExe, &["crt1.o"]),
+    ]);
```

Comment:
Not sure which libc flavor is used on wasm32-wasi, just kept the status quo behavior.
cc @alexcrichton

Reply:
I think this is correct. LLVM has a notion of PIC for wasm but we don't currently bind/expose that in Rust. We can tweak this as necessary if we get around to exposing that.

Reply:
Note that the `crt1.o` object is not added if we are building an executable linking to libc dynamically or a cdylib (looks like both are supported by the wasm32-wasi target).
With any of non-wasm libc flavors that wouldn't be correct.